internal/support/scheduler/init.go

```diff
@@ -0,0 +1,265 @@
+var schedulerClient scheduler.SchedulerClient
+
+func Retry(useConsul bool, useProfile string, timeout int, wait *sync.WaitGroup, ch chan error) {
+	now := time.Now()
+	until := now.Add(time.Millisecond * time.Duration(timeout))
+	for time.Now().Before(until) {
+		var err error
+		//When looping, only handle configuration if it hasn't already been set.
+		if Configuration == nil {
+			Configuration, err = initializeConfiguration(useConsul, useProfile)
+			if err != nil {
+				ch <- err
+				if !useConsul {
+					//Error occurred when attempting to read from local filesystem. Fail fast.
+					close(ch)
+					wait.Done()
+					return
+				}
+			} else {
+				//Check against boot timeout default
+				if Configuration.Service.BootTimeout != timeout {
+					until = now.Add(time.Millisecond * time.Duration(Configuration.Service.BootTimeout))
+				}
+				// Setup Logging
+				logTarget := setLoggingTarget()
+				LoggingClient = logger.NewClient(internal.SupportSchedulerServiceKey, Configuration.Logging.EnableRemote, logTarget)
+			}
+		}
+
+		if Configuration != nil {
+			break
+		}
+		time.Sleep(time.Second * time.Duration(1))
+	}
+	close(ch)
+	wait.Done()
+
+	return
+}
+
+func Init(useConsul bool) bool {
+	if Configuration == nil {
+		return false
+	}
+
+	if useConsul {
+		chConfig = make(chan interface{})
+		go listenForConfigChanges()
+	}
+	return true
+}
+
+func Destruct() {
+	if chConfig != nil {
+		close(chConfig)
+	}
+}
+
+func initializeConfiguration(useConsul bool, useProfile string) (*ConfigurationStruct, error) {
+	//We currently have to load configuration from filesystem first in order to obtain ConsulHost/Port
+	conf := &ConfigurationStruct{}
+	err := config.LoadFromFileV2(useProfile, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	if useConsul {
+		conf, err = connectToConsul(conf)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return conf, nil
+}
+
+func connectToConsul(conf *ConfigurationStruct) (*ConfigurationStruct, error) {
+	//Obtain ConsulConfig
+	cfg := consulclient.NewConsulConfig(conf.Registry, conf.Service, internal.SupportSchedulerServiceKey)
+
+	// Register the service in Consul
+	err := consulclient.ConsulInit(cfg)
+
+	if err != nil {
+		return conf, fmt.Errorf("connection to Consul could not be made: %v", err.Error())
+	}
+	// Update configuration data from Consul
+	updateCh := make(chan interface{})
+	errCh := make(chan error)
+	dec := consulclient.NewConsulDecoder(conf.Registry)
+	dec.Target = &ConfigurationStruct{}
+	dec.Prefix = internal.ConfigV2Stem + internal.SupportSchedulerServiceKey
+	dec.ErrCh = errCh
+	dec.UpdateCh = updateCh
+
+	defer dec.Close()
+	defer close(updateCh)
+	defer close(errCh)
+	go dec.Run()
+
+	select {
+	case <-time.After(2 * time.Second):
+		err = errors.New("timeout loading config from registry")
+	case ex := <-errCh:
+		err = errors.New(ex.Error())
+	case raw := <-updateCh:
+		actual, ok := raw.(*ConfigurationStruct)
+		if !ok {
```

Comment:
In order for the scheduler to pull its configuration from Consul, you need to make config-seed aware of the service. I don't see that logic in this PR. Do you intend to include it? If so, you will need to add the respective service key constant to the list found here:
https://github.com/edgexfoundry/edgex-go/blob/master/internal/seed/config/populate.go#L188

Reply:
I added the service key entry as well as changed the array bounds.